src/main/java/org/jitsi/srtp/crypto/Aes.java

```diff
@@ -599,77 +756,139 @@ else if (simpleClassName.contains("."))
     }
 
     /**
-     * Implements <tt>BlockCipherFactory</tt> using BouncyCastle.
+     * Implements {@link CipherFactory} using Jitsi SRTP's OpenSSL.
+     */
+    public static class OpenSSLCipherFactory
+        extends CipherFactory
+    {
+        public OpenSSLCipherFactory()
+        {
+            super(new JitsiOpenSslProvider());
+        }
+
+        private boolean trySuperApi = true;
+        private Constructor<Cipher> cipherConstructor;
+        private Field cipherProviderField;
+
+        private synchronized void getMethods()
```

Comment:
Instead of this reflection hack, couldn't we do the same as you did with the HmacWrapper? There's a protected constructor `javax.crypto.Cipher#Cipher(javax.crypto.CipherSpi, java.security.Provider, java.lang.String)`.

Reply:
I looked at doing that -- sadly that constructor does
```
        if (!JceSecurityManager.INSTANCE.isCallerTrusted(provider)) {
            throw new IllegalArgumentException("Cannot construct cipher");
        }
```
as its first operation.  (In JDK 8 it's a `NullPointerException` rather than an `IllegalArgumentException`, but the code is otherwise the same.)

Reply:
Ah, so this is the JAR check. I only saw the NPE and misread it for something else. Too bad.